Make PHP 7.2 the minimum required version for 4.x

markstory opened the pull request:
After a healthy discussion the consensus seems to be that requiring 7.2 is reasonable given that 7.1 will soon be end of life and that all major distros already have PHP7.2 or will very soon.

Refs #13103

markstory committed:
Make PHP 7.2 the minimum required version for 4.x

After a healthy discussion the consensus seems to be that requiring 7.2
is reasonable given that 7.1 will soon be end of life and that all major
distros already have PHP7.2 or will very soon.

Refs #13103

markstory committed:
Update scrutinizer as well

ADmad commented on .travis.yml:
```diff
@@ -48,12 +42,7 @@ before_install:
   - phpenv config-rm xdebug.ini
 
   - if [ $DB = 'mysql' ]; then mysql -u root -e 'CREATE DATABASE cakephp_test;'; fi
-  - if [ $DB = 'mysql' ]; then mysql -u root -e 'CREATE DATABASE cakephp_test2;'; fi
```
The additional databases aren't needed anymore?

markstory replied:
Nope. I don't think we ever used them even in 3.x

ADmad replied:
Huh, I always thought they were for cross db tests.

markstory replied:
They were, but I don't think we've had those in Cake 3.

ADmad commented:
phpunit should also be bumped to v8 which is PHP 7.2+. Sticking with an older version could bite use in the future, we already had enough trouble support both phpunit 5.7 and 6 for Cake 3.

markstory committed:
Bump phpunit

ADmad commented:
Zoinks! Updating to phpunit v8 broke apart our test suite :stuck_out_tongue:.

markstory committed:
Don't update PHPUnit.

It makes our test suite really sad as methods in PHPUnit have return
types that make our tests very sad.

markstory commented:
> Fatal error: Declaration of Cake\TestSuite\TestCase::setUp() must be compatible with PHPUnit\Framework\TestCase::setUp(): void in /home/travis/build/cakephp/cakephp/src/TestSuite/TestCase.php on line 36

Going to rollback the PHPUnit version until we can make a separate effort to do that.

markstory commented:
Appveyor is out to lunch again, but otherwise builds look good. Are there any other changes that need to be made right now?

lorenzo commented:
Looks good to me
